File: src/main/java/edu/berkeley/veloxms/storage/TachyonUtils.java
package edu.berkeley.veloxms.storage;

import java.nio.ByteBuffer;
import java.io.IOException;


/**
 * This class contains some static methods to make interacting with Tachyon
 * easier.
 */
public class TachyonUtils {


    private TachyonUtils() {
        // private constructor, do nothing
    }

    public static byte[] long2ByteArr(long id) {
        ByteBuffer key = ByteBuffer.allocate(8);
        key.putLong(id);
        return key.array();
    }

}

File: veloxms-core/src/main/scala/edu/berkeley/veloxms/storage/TachyonUtils.scala
package edu.berkeley.veloxms.storage

import java.nio.ByteBuffer
import java.io.IOException
import scala.util.{Try, Success, Failure}


/**
 * This class contains some static methods to make interacting with Tachyon
 * easier.
 */
object TachyonUtils {

    def long2ByteArr(id: Long): Array[byte] = {
        val key = ByteBuffer.allocate(8)
        key.putLong(id).array()

    }

    // could make this a z-curve key instead
    def twoDimensionKey(key1: Long, key2: Long): Array[byte] = {
        val key = ByteBuffer.allocate(16)
        key.putLong(key1)
        key.putLong(key2)
        key.array()
    }

    def getStore(url: String): Try[ClientStore] = {
        Try(ClientStore.getStore(new TachyonURI(url)))
    }


}
